frontend/lib/resolution/scope-queries.cpp

```diff
@@ -1061,16 +1134,23 @@ static const Scope* nextHigherScope(Context* context, const Scope* scope) {
 // my performance benchmarks they didn't occur frequently so were not
 // worth the additional checking by this function. This list is not intended
 // to be complete; rather, it is intended to cover the most common cases.
+//
+// *also* this list is consulted to avoid finding a parent module when
+// that would potentially cause a problem.
 static bool isReservedIdentifier(UniqueString name) {
   static std::unordered_set<UniqueString> reserved = {
+    USTR("atomic"),
     USTR("bool"),
     USTR("complex"),
     USTR("domain"),
     USTR("int"),
+    USTR("locale"),
     USTR("nil"),
-    USTR("uint"),
     USTR("real"),
+    USTR("sparse"),
     USTR("string"),
+    USTR("subdomain"),
+    USTR("uint"),
     USTR("void"),
   };
```

Comment:
Yes. We have the existing issue #19197 about problems in this area and this PR is doing enough without trying to solve that comprehensively. For this PR, I added cases to this list as-needed to resolve tests that were failing. (Of course, I did not add things to this list that weren't keywords; so occasionally I needed to adjust some tests to avoid warnings about surprising shadowing in a different way).

>  I ran into problems with certain files named a keyword, so this uses isReservedIdentifier to ignore module name lookups for implicit modules with those names.